rest/data/interface.go

```diff
@@ -73,6 +73,11 @@ type Connector interface {
 	FindProjects(string, int, int, bool) ([]model.ProjectRef, error)
 	// FindProjectByBranch is a method to find the projectref given a branch name.
 	FindProjectByBranch(string) (*model.ProjectRef, error)
+
+	// Create/Update project methods
+	CreateProject(*restModel.APIProjectRef) (*model.ProjectRef, error)
+	UpdateProject(*restModel.APIProjectRef) (*model.ProjectRef, error)
```

Comment:
unless there's a really compelling reason (like "it's really hard,") I might be inclined to return restmodels here.